artiq/device_db_testing/kasli_test.py

```diff
@@ -0,0 +1,44 @@
+from artiq.gateware.targets.kasli import (_MasterBase, _SatelliteBase, _dio,
+    _urukul)
+
+from eems import *
+
+class Master(_MasterBase):
+    def __init__(self, *args, **kwargs):
+        _MasterBase.__init__(self, *args, **kwargs)
+        platform = self.platform
+
+        # EEM clock fan-out from Si5324, not MMCX
+        try:
+            self.comb += platform.request("clk_sel").eq(1)
+        except ConstraintError:
+            pass
+
+        devices_config = dict()
+        devices_config.update(eem_dio(self, "eem0", name="dio0"))
+        devices_config.update(eem_urukul(self, "eem3", "eem4", name="urukul0"))
+        check_valid_config(devices_config)
+
+        rtio_channels = get_rtio_channels(devices_config)
+        self.config["HAS_RTIO_LOG"] = None
+        self.config["RTIO_LOG_CHANNEL"] = len(rtio_channels)
+        rtio_channels.append(rtio.LogChannel())
+
+        self.add_rtio(rtio_channels)
+
+        cleanup_devices_config(devices_config)
```

Comment:
I guess that depends on whether in the future there might ever be other functions that use devices_config, but have a problem with non-serialisable entries. If those function are called before `devices_config_to_file`, they would also need extra code for filtering. I guess right now it would be more concise to have it all in `devices_config_to_file`, so I'll merge the two.